avocado/utils/datadrainer.py

```diff
@@ -0,0 +1,105 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2019
+# Author: Cleber Rosa <crosa@redhat.com>
+
+"""
+data drainer
+
+This module provides utility classes for draining data and dispatching
+it to different destinations.  This is intended to be used
+concurrenly with other code, usually test code producing the output
+to be drained/processed.  A thread is started and maintained on behalf
+of the user.
+"""
+
+import threading
+
+
+class BaseDrainer:
+    """
+    Base drainer, doesn't provide complete functionality to be useful.
```

Comment:
Yet, you allow it to be instantiated... shouldn't this be an abstract class?

Reply:
It should... something was bothering me to make this an ABC, but I kind of ignored it.  Changing it in a v2.